pages/micro-frontend/integration.page.tsx

```diff
@@ -21,14 +21,12 @@ interface MicroFrontendWrapperProps {
 
 function MicroFrontendWrapper({ content }: MicroFrontendWrapperProps) {
   const ref = useRef(null);
-  const mountedRef = useRef(false);
 
   useLayoutEffect(() => {
-    if (mountedRef.current) {
-      return;
-    }
-    mountedRef.current = true;
-    createRoot(ref.current!).render(content);
+    const root = createRoot(ref.current!);
+    root.render(content);
+
+    return () => root.unmount();
```

Comment:
This actually leads to React warnings:

```
Warning: Attempted to synchronously unmount a root while React was already rendering. React cannot finish unmounting the root until the current render has completed, which may lead to a race condition.
```

That is probably fine for a demonstration page but I wonder if there are other simple ways of mimic the micro-frontend environment.